packages/components/src/types/sitemap.ts

```diff
@@ -19,8 +20,18 @@ interface IsomerBaseSitemap {
 }
 
 interface IsomerPageSitemap extends IsomerBaseSitemap {
-  layout: Exclude<IsomerPageLayoutType, "file" | "link">
+  layout: Exclude<IsomerPageLayoutType, "collection" | "file" | "link">
 }
+
+export interface IsomerCollectionPageSitemap extends IsomerBaseSitemap {
+  layout: Extract<IsomerPageLayoutType, "collection">
+  // TODO: Reconsider how this is done as currently every item in the sitemap has the same props
```

Comment:
thought: I thought that this setup makes sense as we can make layout-specific sitemap entries that's still type-safe, so we can extract these extra props if we detect the sitemap entry layout is different.

No action required for now, just curious about this TODO what are we able to potentially do